oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/ElasticRequestHandler.java

```diff
@@ -191,8 +193,6 @@ public BoolQuery.Builder baseQueryBuilder() {
                     } else {
                         bqb.must(m -> m.moreLikeThis(mltQuery(mltParams)));
                     }
-                } else {
-                    bqb.must(m -> m.bool(similarityQuery(queryNodePath, sp)));
```

Comment:
the knn query should be part of the base query to be sure all scenarios are covered.
This documentation here explains how to do that
https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-knn-query.html

Reply:
I tried to add `knn` as part of base bool query, but following error is printed: 
`Corresponding ES query Query: {"bool":{"filter":[{"exists":{"field":"sim:fv"}}],"should":[{"knn":{"field":"sim:fv","query_vector":[0.1166586503386497... :: ES Response ErrorCause: {"type":"x_content_parse_exception","reason":"[1:125] [bool] failed to parse field [should]","caused_by":{"type":"illegal_argument_exception","reason":"[knn] queries cannot be provided directly, use the [knn] body parameter instead"},"root_cause":[{"type":"x_content_parse_exception","reason":"[1:125] [bool] failed to parse field [should]"}]} : closing scanner, notifying listeners`
Using wrapper for base64-encoded kin query string also results in error. Any hint?